sonar-orchestrator/src/main/java/com/sonar/orchestrator/server/ServerInstaller.java

```diff
@@ -149,19 +151,30 @@ private void copyJdbcDriver(File sonarHome) {
     }
   }
 
-  private static void removeBundledPlugins(File homeDir) {
+  private static void removeBundledPlugins(File homeDir, List<String> bundledPluginsToKeep) {
     LOG.info("Remove bundled plugins");
-    cleanDirectory(new File(homeDir, "lib/bundled-plugins"));
+    cleanDirectory(new File(homeDir, "lib/bundled-plugins"), bundledPluginsToKeep);
     // plugins are bundled in extensions/plugins since version 7.2
-    cleanDirectory(new File(homeDir, "extensions/plugins"));
+    cleanDirectory(new File(homeDir, "extensions/plugins"), bundledPluginsToKeep);
     // SonarSource plugins are bundled in lib/extension since version 8.5
-    cleanDirectory(new File(homeDir, "lib/extensions"));
+    cleanDirectory(new File(homeDir, "lib/extensions"), bundledPluginsToKeep);
   }
 
-  private static void cleanDirectory(File dir) {
+  private static void cleanDirectory(File dir, List<String> bundledPluginsToKeep) {
     try {
-      if (dir.exists()) {
-        FileUtils.cleanDirectory(dir);
+      if (!dir.exists()) {
+        return;
+      }
+      try (Stream<Path> files = Files.list(dir.toPath())) {
+        files.forEach(file -> {
+          if (bundledPluginsToKeep.stream().noneMatch(prefix -> file.getFileName().toString().startsWith(prefix))) {
```

Comment:
this logic in the if statement could be moved to a separate method to increase readability

Reply:
done!